modules/calypso-redirects.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+if ( ! class_exists( 'Jetpack_Calypso_Redirects' ) ) :
+
+class Jetpack_Calypso_Redirects {
+
+	public function __construct() {
+		if ( ! get_option( 'jetpack_calypso_redirects_active' ) || ! is_admin() ) {
+			return;
+		}
+
+		add_action( 'admin_menu', array( $this, 'menu_redirects' ), 30 );
+		add_action( 'admin_enqueue_scripts', array( $this, 'enqueue_scripts' ) );
+
+		add_filter( 'allowed_redirect_hosts', array( $this, 'allow_calypso_domain' ), 10, 1 );
+
+		add_action( 'load-toplevel_page_calypso_plugins', array( $this, 'plugins_redirect' ) );
+
+		if ( current_user_can( 'list_users' ) ) {
+			$this->register_user_redirect_handlers();
+		}
+	}
+
+	private function register_user_redirects() {
+		remove_submenu_page( 'users.php', 'users.php' );
+		remove_submenu_page( 'users.php', 'user-new.php' );
+		remove_submenu_page( 'users.php', 'profile.php' );
+
+		// replace each menu item one by one with its redirect
+		add_submenu_page( 'users.php', __( 'All Users' ), __( 'All Users' ), 'list_users', 'calypso_users', array( $this, 'users_redirect' ) );
+		add_submenu_page( 'users.php', __( 'Add New' ), __( 'Add New' ), 'promote_users', 'calypso_users_new', array( $this, 'users_new_redirect' ) );
+		add_submenu_page( 'users.php', __( 'Your Profile' ), __( 'Your Profile' ), 'read', 'calypso_users_profile', array( $this, 'users_profile_redirect' ) );
+	}
+
+	private function register_user_redirect_handlers() {
+		add_action( 'load-users_page_calypso_users', array( $this, 'users_redirect' ) );
+		add_action( 'load-users_page_calypso_users_new', array( $this, 'users_new_redirect' ) );
+		add_action( 'load-users_page_calypso_users_profile', array( $this, 'users_profile_redirect' ) );
+	}
+
+	private function register_track_event( $event ) {
+		// record in tracks
+		jetpack_require_lib( 'tracks/client' );
+		jetpack_tracks_record_event( wp_get_current_user(), $event );
```

Comment:
If you instead used `record_user_event()`, you could avoid having to pass in the current user, and it would be easier to pass data along with props, which I think makes a cleaner way to discover this data.  For example

```
JetpackTracking::record_user_event( 'jetpack_admin_calypso_redirect', array(
				'menu_item' => $menu_item (example "users_add_new")
			) );
```

cc @withinboredom for opinion on this

Reply:
I don't think you even need the `jetpack_` prefix, IIRC, it gets added automagically. Record user event is better to send the correct user agent, etc.